tests/build/_internal/test_cmake_file.py

```diff
@@ -7,60 +7,51 @@
 from tests.build._internal.config.factories import ConfigFactory
 from mbed_tools.build._internal.cmake_file import generate_mbed_config_cmake_file, _render_mbed_config_cmake_template
 
+@pytest.fixture()
+def fake_target():
+    target = dict()
+    target["labels"] = ["foo"]
+    target["extra_labels"] = ["morefoo"]
+    target["features"] = ["bar"]
+    target["components"] = ["baz"]
+    target["macros"] = ["macbaz"]
+    target["device_has"] = ["stuff"]
+    target["c_lib"] = ["c_lib"]
+    target["core"] = ["core"]
+    target["printf_lib"] = ["printf_lib"]
+    target["supported_form_factors"] = ["arduino"]
+    target["supported_c_libs"] = {"loo": ["ginormous"]}
+    target["supported_application_profiles"] = ["full", "bare-metal"]
+    yield target
+
 
 class TestGenerateCMakeListsFile:
-    def test_correct_arguments_passed(self):
-        target = dict()
-        target["labels"] = ["foo"]
-        target["extra_labels"] = ["morefoo"]
-        target["features"] = ["bar"]
-        target["components"] = ["baz"]
-        target["macros"] = ["macbaz"]
-        target["device_has"] = ["stuff"]
-        target["c_lib"] = ["c_lib"]
-        target["core"] = ["core"]
-        target["printf_lib"] = ["printf_lib"]
-        target["supported_form_factors"] = ["arduino"]
+    def test_correct_arguments_passed(self, fake_target):
         config = ConfigFactory()
         mbed_target = "K64F"
-        toolchain_name = "GCC"
-        target["supported_c_libs"] = {toolchain_name.lower(): ["small", "std"]}
-        target["supported_application_profiles"] = ["full", "bare-metal"]
+        toolchain_name = "loo"
 
-        result = generate_mbed_config_cmake_file(mbed_target, target, config, toolchain_name)
+        result = generate_mbed_config_cmake_file(mbed_target, fake_target, config, toolchain_name)
 
-        assert result == _render_mbed_config_cmake_template(target, config, toolchain_name, mbed_target,)
+        assert result == _render_mbed_config_cmake_template(fake_target, config, toolchain_name, mbed_target,)
 
 
 class TestRendersCMakeListsFile:
-    def test_returns_rendered_content(self):
-        target = dict()
-        target["labels"] = ["foo"]
-        target["extra_labels"] = ["morefoo"]
-        target["features"] = ["bar"]
-        target["components"] = ["baz"]
-        target["macros"] = ["macbaz"]
-        target["device_has"] = ["stuff"]
-        target["core"] = ["core"]
-        target["c_lib"] = ["c_lib"]
-        target["printf_lib"] = ["printf_lib"]
-        target["supported_form_factors"] = ["arduino"]
+    def test_returns_rendered_content(self, fake_target):
         config = ConfigFactory()
-        toolchain_name = "baz"
-        target["supported_c_libs"] = {toolchain_name.lower(): ["small", "std"]}
-        target["supported_application_profiles"] = ["full", "bare-metal"]
-        result = _render_mbed_config_cmake_template(target, config, toolchain_name, "target_name")
+        toolchain_name = "loo"
```

Comment:
Why did we change the name of this?

Reply:
There wasn't a name that both tests used, and it is required that this name match what's in the fixture for the fake_target. Any better ideas on how to match names? I don't like repeating the name.

Reply:
Could we define a `toolchain_name` fixture that just returns the name as a string?

Reply:
I made a global. :/